pkg/operator/ceph/cluster/mgr/dashboard.go

```diff
@@ -104,7 +100,13 @@ func (c *Cluster) configureDashboardModules() error {
 	}
 	if hasChanged {
 		logger.Info("dashboard config has changed. restarting the dashboard module")
-		return c.restartDashboard()
+		if err := c.restartDashboard(); err != nil {
+			return errors.Wrap(err, "failed to restart dashboard")
+		}
+	}
+	_, err := c.initializeSecureDashboard()
```

Comment:
It looks like `initializeSecureDashboard()` always returns false. In that case, please update that method to just return the err instead of ignoring the return value.  
```suggestion
	err := c.initializeSecureDashboard()
```